staging/src/k8s.io/client-go/tools/cache/delta_fifo.go

```diff
@@ -129,8 +133,27 @@ type DeltaFIFO struct {
 	// emitDeltaTypeReplaced is whether to emit the Replaced or Sync
 	// DeltaType when Replace() is called (to preserve backwards compat).
 	emitDeltaTypeReplaced bool
+
+	// Called with every object if non-nil.
+	transformer TransformFunc
 }
 
+// TransformFunc allows for transforming an object before it will be processed.
+// TransformFunc (similarly to ResourceEventHandler functions) should be able
+// to correctly handle the tombstone of type cache.DeletedFinalStateUnknown.
+//
+// New in v1.27: In such cases, the contained object will already have gone
+// through the transform object separately (when it was added / updated prior
+// to the delete), so the TransformFunc can likely safely ignore such objects.
```

Comment:
Nice, this comment is very good!

Should we maybe rephrase this to state that "ignore" means returning the input object to be 100% clear?

Reply:
Yes, I will update. I'm adding a test now

Reply:
Nice, thanks! Also feel free to take some logic from https://github.com/kubernetes/kubernetes/pull/116432 as well, or we can do that as a follow up after this, thats up to you